packages/react-router-dom/dom.ts

```diff
@@ -97,13 +97,18 @@ export function getSearchParamsForLocation(
   let searchParams = createSearchParams(locationSearch);
 
   if (defaultSearchParams) {
-    for (let key of defaultSearchParams.keys()) {
+    // Use `defaultSearchParams.forEach(...)` here instead of iterating of
+    // `defaultSearchParams.keys()` to work-around a bug in Firefox related to
+    // web extensions. Relevant Bugzilla tickets:
+    // https://bugzilla.mozilla.org/show_bug.cgi?id=1414602
+    // https://bugzilla.mozilla.org/show_bug.cgi?id=1023984
+    defaultSearchParams.forEach((key) => {
```

Comment:
Isn't the signature reversed here and `key` would be the second parameter?

<img width="496" alt="Screenshot 2023-06-21 at 9 59 00 AM" src="https://github.com/remix-run/react-router/assets/1609022/b0f20aa9-beee-4ca5-be70-51a9f5a3fddf">

This also tells me we've not got sufficient test coverage over this, so if you're interested in trying to add a test to around this against `useSearchParams` I'd like to get one added before we merge this. Otherwise I'll try to get some tests wired up when I have a chance.